### PR TITLE
19w03a mappings

### PR DESCRIPTION
**Pull request by Runemoro**

### CODE REVIEW

File: mappings/net/minecraft/recipe/RecipeType.mapping
Comment (asiekierka) [inline mappings/net/minecraft/recipe/RecipeType.mapping]: You sure that's a good name?

File: mappings/net/minecraft/recipe/RecipeSerializer.mapping
Comment (asiekierka) [inline mappings/net/minecraft/recipe/RecipeSerializer.mapping:1]: I've done a few changes already, make sure to take them into account.

File: mappings/CraftingRecipe.mapping
Comment (asiekierka) [inline mappings/CraftingRecipe.mapping]: No package name? :thinking:

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/CraftingRecipe.mapping
Comment (Runemoro) [reply mappings/CraftingRecipe.mapping]: Oops, I'll fix that in a few minute.

File: mappings/net/minecraft/recipe/RecipeType.mapping
Comment (Runemoro) [reply mappings/net/minecraft/recipe/RecipeType.mapping]: It returns `Optional.of(recipe)` (`recipe` is the first parameter)  if the recipe matches, of `Optional.empty()` otherwise. Any suggestion for a different name?

**Comment by SquidDev** [comment]
On a side note, I don't know if it's worth renaming the more specialised versions of `craft`/`matches` in subclasses of `Recipe`? I don't know how often people access them directly.

**Comment by asiekierka** [comment]
Shouldn't unless necessary - the toolchain is supposed to, but is not, as they are generic obfuscation artifacts :thinking:

File: mappings/CraftingRecipe.mapping
Comment (Runemoro) [reply mappings/CraftingRecipe.mapping]: Fixed

File: mappings/net/minecraft/recipe/RecipeSerializer.mapping
Comment (Runemoro) [reply mappings/net/minecraft/recipe/RecipeSerializer.mapping:1]: Ok. I renamed `RecipeSerializers` -> `RecipeSerializer` as it's an interface all serializers implement, not just the place where serialziers are stored.

File: mappings/net/minecraft/recipe/RecipeType.mapping
Comment (SquidDev) [reply mappings/net/minecraft/recipe/RecipeType.mapping]: I wonder if `get` on its own is sufficient? The type should indicate that it only returns if it does match.

Comment (Runemoro) [reply mappings/net/minecraft/recipe/RecipeType.mapping]: You're right, that's better. I changed it to `get`.

**Review by SquidDev — APPROVED** [review]
Thank you for putting this together <3.

File: mappings/net/minecraft/block/ComposterBlock.mapping
Comment (asiekierka) [inline mappings/net/minecraft/block/ComposterBlock.mapping]: If field, "dirty"

File: mappings/net/minecraft/block/ComposterBlock.mapping
Comment (asiekierka) [inline mappings/net/minecraft/block/ComposterBlock.mapping]: Likewise

File: mappings/net/minecraft/client/util/EntityHitResult.mapping
Comment (asiekierka) [inline mappings/net/minecraft/client/util/EntityHitResult.mapping]: If the class isn't envtype-client-only, util/

File: mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping
Comment (asiekierka) [inline mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: Not really the toast icon; more of a sub-icon showing the "machine" used.

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/util/EntityHitResult.mapping
Comment (Runemoro) [reply mappings/net/minecraft/client/util/EntityHitResult.mapping]: No, it's both sides.

File: mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping
Comment (Runemoro) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: Changed it to `getMachineItem`

File: mappings/net/minecraft/client/util/EntityHitResult.mapping
Comment (Runemoro) [reply mappings/net/minecraft/client/util/EntityHitResult.mapping]: Oh, sorry, I misread what you said. Didn't realize I put it in client, I'll fix that now.

File: mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping
Comment (asiekierka) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: "machine" is still a lousy name as it doesn't make sense in the Minecraft context. And stick to Icon, maybe? Anyone got any ideas?

Comment (SquidDev) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: JEI used to call it crafting type item, and now calls it a recipe catalyst. Not sure if either is entirely accurate here though.

Comment (Runemoro) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: `recipeTypeIcon`?

Comment (liach) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: `getCatalyst` `getTool` `getProcessor`

Comment (Runemoro) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: @liach I think `getRecipeTypeIcon` is clearer than those suggestions. I wouldn't know what the "catalyst" of a recipe is just by the name. I wouldn't really call a crafting table or furnace a "tool".

Comment (liach) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: you can keep it as it is.

File: mappings/net/minecraft/recipe/SpecialRecipeSerializer.mapping
Comment (liach) [inline mappings/net/minecraft/recipe/SpecialRecipeSerializer.mapping:1]: `BuiltInRecipeSerializer`?

File: mappings/net/minecraft/recipe/cooking/CookingRecipeSerializer.mapping
Comment (liach) [inline mappings/net/minecraft/recipe/cooking/CookingRecipeSerializer.mapping:2]: `Factory`?

Comment (liach) [inline mappings/net/minecraft/recipe/cooking/CookingRecipeSerializer.mapping:2]: `Factory`?

File: mappings/net/minecraft/recipe/SpecialRecipeSerializer.mapping
Comment (Runemoro) [reply mappings/net/minecraft/recipe/SpecialRecipeSerializer.mapping:1]: Mojang calls them "special recipes": `crafting_special_bookcloning`, `crafting_special_mapcloning`, `crafting_special_firework_rocket`, etc.

Comment (liach) [reply mappings/net/minecraft/recipe/SpecialRecipeSerializer.mapping:1]: Hmm. I thought "builtin" because these recipes are totally defined by code just like vanilla chest models are.

Comment (asiekierka) [reply mappings/net/minecraft/recipe/SpecialRecipeSerializer.mapping:1]: Special is fine

File: mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping
Comment (asiekierka) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: It's just that we have a class called RecipeType.

Comment (Runemoro) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: Right now the vanilla icons depend only on the recipe type. But to avoid confusion, maybe it could be named `getRecipeSortIcon`/`getRecipeKindIcon`?

Comment (asiekierka) [reply mappings/net/minecraft/recipe/cooking/SmokingRecipe.mapping]: getRecipeKindIcon sounds alright IMO if not 100% perfect

**Review by asiekierka — APPROVED** [review]
Get a second review and ping me so we can ship this ASAP!

**Comment by liach** [comment]
lgtm

**Pull request merged**
